docs/advanced-topics/session-management.md

```diff
@@ -0,0 +1,26 @@
+---
+id: session-management
+title: Session Management
+---
+In a multiplayer game, we might want to have some session management to allow players to correctly reconnect to the game. What this exactly means might vary depending on the game, but the core idea is that there is some data that needs to be kept when a player disconnects, and assigned back to them when they reconnect. We must also make sure that the data is reassigned to the right player.
+
+## Linking data to players
+To reassign data to the right player when they reconnect, we need a way to link this data to the player. The clientId generated by NGO cannot be used, however, since it is generated when a player connects and is disposed of when they disconnect. It can then be assigned to a different player when if they connect to the server after the first one disconnected. To properly link data to a specific player, we need an ID that is not tied to the current session, that persists through time, and that is unique to that player. A few solutions can be used, such as a login system with unique player accounts, or simply a Globally Unique Identifier (GUID). A GUID can be generated via `System.Guid.NewGuid()` and then saved to the `PlayerPrefs`, for example.
+
+When we have such a way to uniquely identify a player, we can then use it to map each player's data that would be needed when reconnecting, such as the current state of their character for example. Then, we need to make sure this data is up to date and to keep it after a player disconnects. We could add a timeout after which we can purge the data or we could decide to clear all data only when a session completes. And finally, once we have up to date data that we can link to a specific player who disconnected earlier, we can now handle reconnection properly.
+
+## Handling reconnection
+In cases where we use a Player Object, a new Default Player Prefab will be automatically spawned when any player connects to the game (including if they are reconnecting). We can then use the player's data to update that object so that its state returns to what it was before disconnecting. In those cases, we would need to keep all the important data that we want to restore and map it to the player using our identification system.
+
+In cases where we use the ownership model, we could instead keep the objects that a player owns when they disconnects, and set the reconnected player as their new owner. To accomplish this, the only data we would need to keep would be the mapping between those objects and their owning player's identifier, then when a player reconnects we could use this mapping to set them as the new owner.
```

Comment:
"ownership model" ? Do we mean Netcode's ownership?

Reply:
Yes, I used the term defined [here](https://docs-multiplayer.unity3d.com/docs/develop/basics/networkobject). Maybe I should add a link to that page at the beginning of this section?

Reply:
In that case, for consistency we should probably rewrite this sentence from that other page: "If you want a client to control multiple objects use the ownership model instead."

Reply:
why do you call this a "model"? To me it's more of an API or a feature. Prediction is a lag compensation "model", there's multiple ways to implement it, but the principle is the same. To me, a model is more like a "template" or a pattern, more than an SDK feature.

Reply:
What I mean is that the term "model" to distinguish between using a PlayerObject vs using manually specified Ownership for NetworkObjects comes from [here](https://docs-multiplayer.unity3d.com/docs/develop/basics/networkobject/index.html). I agree with what you are saying, that another word might be less confusing, but we should also change it in that other page if we want to be consistant

Reply:
@s-omeilia-unity ^^^

Reply:
Would "approach" be a better term for that distinction?